cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -74,14 +74,23 @@ func cleanup(indexDir string, repos []string, now time.Time) {
 		shardsLog(indexDir, "restore", shards)
 	}
 
-	// index: Move non-existant repos into trash
-	for _, shards := range index {
+	// index: Move non-existent repos into trash
+	for repo, shards := range index {
 		// Best-effort touch. If touch fails, we will just remove from the
 		// trash sooner.
 		for _, shard := range shards {
 			_ = os.Chtimes(shard.Path, now, now)
 		}
 
+		if _, err := os.Stat(filepath.Join(indexDir, zoekt.TombstoneFileName)); err == nil {
```

Comment:
it feels somewhat wasteful to be doing this syscall inside of the for loop. Probably adds up to nothing, but I can't help but suggest to hoist this check out of the loop.